src/docfx/restore/RestoreGit.cs

```diff
@@ -27,21 +27,40 @@ public static async Task Restore(string docsetPath, Config config, Func<string,
                 group (git.branch, git.flags)
                 by git.remote;
 
-            await ParallelUtility.ForEach(gitDependencies, RestoreGitRepo, Progress.Update);
+            var workTrees = new List<string>();
+
+            await ParallelUtility.ForEach(gitDependencies, async group => { workTrees.AddRange(await RestoreGitRepo(group)); }, Progress.Update);
+
+            foreach (var workTree in workTrees)
+            {
+                // update the last write time
+                Directory.SetLastWriteTimeUtc(workTree, DateTime.UtcNow);
+            }
 
             if (!isDependencyRepo && LocalizationUtility.TryGetContributionBranch(docsetPath, out var contributionBranch, out var repo))
             {
                 await GitUtility.Fetch(repo.Path, repo.Remote, contributionBranch, config);
             }
 
-            async Task RestoreGitRepo(IGrouping<string, (string branch, GitFlags flags)> group)
+            async Task<List<string>> RestoreGitRepo(IGrouping<string, (string branch, GitFlags flags)> group)
```

Comment:
nit: how about set modified time inside, so you don't have the change the return value and structure that much.